enumerables.rb

```diff
@@ -0,0 +1,141 @@
+=begin
+    
+rescue => exception
+    practicing RUBY ENUMERABLES by building my own methods
+=end
+
+module Enumerable
+
+    #  1- Create #my_each, a method that is identical to #each but (obviously) does 
+    #     not use #each. You’ll need to remember the yield statement. Make sure it
+    #     returns the same thing as #each as well.
+    def my_each
+      for i in 0...self.size
+        yield(self[i])
+      end
+      return self
+    end
+
+    #  2- Create #my_each_with_index in the same way.
+    def my_each_with_index 
+      for i in 0...self.size
+        yield(self[i], i)
+      end
+      return self
+    end
+
+    #  3-Create #my_select in the same way, though you may use #my_each in your definition (but not #each).
+    def my_select
+      arr = []
+      self.my_each do |item|
+          arr << item if yield(item)
+      end
+      return arr
+    end
+
+    # 4 - Create #my_all?
+    def my_all?
+      self.my_each do |item|
+        return false unless yield(item)
+      end
+      return true
+    end 
+
+    # 5 - Create #my_any?
+    def my_any?
+      self.my_each do |item|
+        return true if yield(item)
+      end
+      return false
+    end 
+
+    # 6 - Create #my_none?
+    def my_none?
+      self.my_any? do |item|
+        return false if yield(item)
+      end
+      return true
+    end
+
+    # 7 - Create #my_count
+    def my_count(arg=nil)
+      if !(arg.nil?)
+        i = 0
+        self.my_each do |item|
+            i += 1 if item == arg
+        end
+        return i
+      elsif block_given?
+        return self.my_select {|item| yield(item)}.length
```

Comment:
Don't use `my_select` or `length`

Reply:
thanks bro for the review..i have resubmitted a proposed correction